### ISSUE TITLE
Tomboy no longer minimizes to system tray / notification area

### ISSUE THREAD

**Issue by HighCommander4** [tomboy-notes/tomboy#84]
Steps to reproduce:
  Press the close button on the main Tomboy window ("Search All Notes").

Expected results:
  Tomboy minimizes to the system tray / notification area, or there is an
  option to make it behave this way.

Actual results:
  Tomboy shuts down altogether. I can't find a preference to control
  this behaviour.

Interestingly, until recently this used to work. I'm not sure why the behaviour changed.

Running Tomboy v1.14.1 on Debian 9.

**Comment by HighCommander4** [comment]
> Interestingly, until recently this used to work. I'm not sure how or why the behaviour changed.

In particular, I'm not seeing any recent updates to the Tomboy package in the [Debian package tracker](https://tracker.debian.org/pkg/tomboy) that might have brought this change of behaviour, making this all the more weird.

**Comment by davidbannon** [comment]
I wonder if rather than see Tomboy shut down, you are seeing a crash.  Starting tomboy from the commandline may give you a hint.  Either way, I'd consider upgrading tomboy. My Ubuntu 16.04 offers Tomboy 1.15.6 in its repo.

**Comment by HighCommander4** [comment]
> Starting tomboy from the commandline may give you a hint.

Curiously, when I start it from the command line, it doesn't open a window at all. It just hangs.

> Either way, I'd consider upgrading tomboy. My Ubuntu 16.04 offers Tomboy 1.15.6 in its repo.

As mentioned, I'm running Debian 9, where the latest version offered is 1.14.1 (I checked the backports repository as well). Do you have a suggestion for where to obtain the newer version from?

**Comment by HighCommander4** [comment]
I should note that this issue happens one of my machines running Debian 9, but not another. So I think it's indeed likely that it's just crashing on the one machine for some reason.

**Comment by HighCommander4** [comment]
This _just_ started happening on another one of my Debian 9 machines. I just upgraded the kernel from 4.9 to 4.14, I wonder if that is related...

**Comment by HighCommander4** [comment]
> I just upgraded the kernel from 4.9 to 4.14, I wonder if that is related...

Doesn't seem that way:

  * Rebooted into 4.9 kernel --> problem is gone
  * Rebooted into 4.14 kernel again --> problem is gone

So... rebooting fixed the problem? But what caused it? Really weird...

**Comment by HighCommander4** [comment]
I should clarify that when the problem occurs, Tomboy does not create a system tray icon at all. So I don't think the problem occurs when closing the window, I think it occurs on startup, when it fails to create the system tray icon for some reason.

**Comment by davidbannon** [comment]
Yep, that sounds likely. Did you try starting tomboy from the command line ?
A possible explanation of why rebooting seemed to help, if there had been a previous crash with Tomboy, maybe, just maybe, some process was still running and that interfered with Tomboy's startup. Tomboy does check to ensure its not a second process although I don't know what mechanism is used. (For Tomboy-ng, I just grep for a running process called 'tomboy-ng' and it can suffer a fate as discussed.)
I do think you need to start Tomboy from command line and add --debug
Davo
